### PR TITLE
Support SXCI - Classic BAdI implementations

### PR DESCRIPTION
**Pull request by christianguenter2**
#156

### CODE REVIEW

**Comment by larshp** [comment]
cool, I'll look through the PRs during the weekend

File: src/zabapgit_object_sxci.prog.abap [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+                 ig_data = lt_function_codes ).
+
+    io_xml->add( iv_name = co_badi_component_name-control_composites
+                 ig_data = lt_control_composites ).
+
+    io_xml->add( iv_name = co_badi_component_name-customer_includes
+                 ig_data = lt_customer_includes ).
+
+    io_xml->add( iv_name = co_badi_component_name-screens
+                 ig_data = lt_screens ).
```
Comment (larshp) [inline src/zabapgit_object_sxci.prog.abap]: can we put lt_filters + lt_function_codes + lt_control_composites + lt_customer_includes + lt_screens into a structure? this way it is easier to extend, plus less code as io_xml->add will only be called once

File: src/zabapgit_object_sxci.prog.abap [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+      EXPORTING
+        iv_name = co_badi_component_name-customer_includes
+      CHANGING
+        cg_data = lt_customer_includes ).
+
+    io_xml->read(
+      EXPORTING
+        iv_name = co_badi_component_name-screens
+      CHANGING
+        cg_data = lt_screens ).
```
Comment (larshp) [inline src/zabapgit_object_sxci.prog.abap]: same above, there will only be 1 read

**Comment by christianguenter2** [comment]
@larshp Good idea. I refactored to one common structure. Please review again.

**Comment by larshp** [comment]
Thanks, looks good, merging

**Pull request merged**